### PR TITLE
Create calendar page layout and set up navigation

### LINKED ISSUE #3: Create the calendar page layout
Create the basic layouts for the calendar page, please refer to the screenshots.
You don't need to have anything for the calendar view itself yet!

- [x] Tabs on the top with text "Calendar" and "Cycle"
- [x] Tabs on the bottom with respective icons and the red "+" button in the middle
Screen Shot 2022-11-01 at 1 50 07 PM

### LINKED ISSUE #4: Implement the navigation between different pages
Implement the navigation behaviour, if a pages doesn't exist yet, make an empty page to test out the navigation.

- [x] "Calendar" and "Cycle" should now take you to an empty page
- [x] You should be able to navigate between "Setting" and "Education" using the icon buttons at the bottom
- [x] The tab at the bottom should be there on all pages, while the tab on top for "Calendar" and "Cycle" should disappear if you are not on the calendar page.

### PR DESCRIPTION
**Pull request by michaelng0107**
Closes #3 and closes #4.

Added common bottom navigation bar to `CommonComponents.kt`
Added Calendar Tab layout to `CalendarScreen.kt`
Created `NavigationGraph.kt` to organize routes.

Notes:
- There were conflicts between `develop` and this branch, and it wasn't possible to have 2 NavHosts, so I moved the routes defined by @kevinle623 to `NavigationGraph.kt`. See commit [b8c78b4caf9cdffb3ae699579e9dc21ffd32784c](https://github.com/uoftblueprint/the-period-purse-2023/pull/25/commits/b8c78b4caf9cdffb3ae699579e9dc21ffd32784c)
- Updated `androidx.test.ext:junit` from `1.1.3` to `1.1.4` to resolve issue seen in [e148cd1536f05b72d60bc9c23dfd94b8e47fea55](https://github.com/uoftblueprint/the-period-purse-2023/pull/25/commits/e148cd1536f05b72d60bc9c23dfd94b8e47fea55)

### CODE REVIEW

File: app/src/androidTest/java/com/example/theperiodpurse/NavigationTest.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    private fun navigateToInfoScreen() {
+        composeTestRule.onNodeWithText(Screen.Learn.name).performClick()
+    }
+
+    @Test
+    fun appNavHost_clickCalendarFABOnSettingsScreen_navigatesToCalendarScreen() {
+        navigateToSettingsScreen()
+        composeTestRule.onNodeWithContentDescription("fab")
```
Comment (michaelng0107) [inline app/src/androidTest/java/com/example/theperiodpurse/NavigationTest.kt]: There might be a better way of getting the FAB node. It doesn't have text on it so I had to use something like contentDescription. We could look into test tags/making contentDescriptions more standardized

**Comment by michaelng0107** [comment]
Don't think the AndroidTests (where the navigation tests are) are running on CI since we're only running `./gradlew test`. Should we also add `./gradlew connectedAndroidTest` to run instrumental tests?

https://developer.android.com/studio/test/command-line#run-tests-with-gradle

**Comment by leowrites** [comment]
@michaelng0107 
> Don't think the AndroidTests (where the navigation tests are) are running on CI since we're only running `./gradlew test`. Should we also add `./gradlew connectedAndroidTest` to run instrumental tests?
> 
> https://developer.android.com/studio/test/command-line#run-tests-with-gradle

Yes, you can add that to the job in the workflow

**Comment by michaelng0107** [comment]
Something we have to think about is whether its okay to change the CI VM from `ubuntu-latest` to `macOS-latest`, and if a 20 minute github action is acceptable.

**Comment by leowrites** [comment]
@michaelng0107 
> Something we have to think about is whether its okay to change the CI VM from `ubuntu-latest` to `macOS-latest`, and if a 20 minute github action is acceptable.

In https://github.com/marketplace/actions/android-emulator-runner, there is a section on reducing startup time through AVD snapshot caching. Maybe that can reduce the 20 minutes?

**Review by leowrites — APPROVED** [review]
Great work, the code is very clean 👍

**Review by elenawzy — APPROVED** [review]

**Pull request merged**